### ISSUE TITLE
UInt Serialization

### ISSUE THREAD

**Issue by aphelionz** [orbitdb/orbitdb#672]
@tabcat and another user brought our attention to this behavior:

```JavaScript
const IPFS = require('ipfs')
const OrbitDB = require('orbit-db')

const ipfs = new IPFS( {
  EXPERIMENTAL: {
    pubsub: true
  }
})

ipfs.on('ready', async () => {
  const orbitdb = await OrbitDB.createInstance(ipfs)
  var docstore = await orbitdb.docs('test')
  const uint = new Uint8Array([1,2,3,4])
  await docstore.put({_id:'test', uint, })
  const cachedUint = docstore.get('test')[0].uint
  await docstore.close()
  docstore = await orbitdb.docs('test')
  await docstore.load()
  const loadedUint = docstore.get('test')[0].uint
  console.log(`uint: ${uint}`)
  console.log(`cachedUint: ${cachedUint}`)
  console.log(`loadedUint: ${loadedUint}`)
    // uint: 1,2,3,4
    // cachedUint: 1,2,3,4
    // loadedUint: [object Object]
})
```

We should figure out a good strategy on how to retrieve binary Uint values to and from Orbit

**Comment by tabcat** [comment]
my main issue with this is that  in the example the cachedUint doesnt equal the loadedUint. I dont think this is unique to Uints but that when someone adds a new entry, im guessing, the cache is storing the plain js object and not the JSON parsed version of the object like in the store. 

json parsing payloads before caching would change this behavior.

**Comment by tabcat** [comment]
https://github.com/orbitdb/orbit-db-store/pull/73

**Comment by haadcode** [comment]
This issue has been resolved >1.0.0. To make sure, and to document it, I added tests for this in https://github.com/orbitdb/orbitdb/pull/1139.
